### DISCUSSION TITLE
secure websocket as transport

### DISCUSSION THREAD

**Discussion by sthustfo1** [emiago/sipgo#61] [Q&A] [answered]
Is it possible with current implementation to send SIP messages over websocket? or rather secure websocket?

**Comment by emiago** [comment, marked as answer]
yes. It is part of supported protocols. This you should see on Readme. 
Consider that if it is a server you have different func to listen on TLS. 
For client all you need is set message transport either by recipient or call `req.SetTransport()`

**Comment by sthustfo1** [comment]
`req.SetTransport()` does help. How can one set the transport via recipient?

Also one follow up query - If the server is using self signed certs, then is there a way to tell the transport to ignore the same? The error I am seeing is below

`Dec 11 12:52:17.350017 FTL Fail to create transaction error="transport<WSS> dial err=tls: failed to verify certificate: x509: “Main” certificate is not standards compliant"`

**Reply by sthustfo1** [reply]
My bad, should have read the doc - `sipgo.WithUserAgenTLSConfig()`
